src/shogun/mathematics/linalg/LinalgNamespace.h

```diff
@@ -129,19 +129,34 @@ T dot(const SGVector<T>& a, const SGVector<T>& b)
 }
 
 /**
- * Method that computes the mean of vectors or matrices.
+ * Method that computes the mean of vectors or matrices composed of complex numbers.
  *
  * @param a SGVector or SGMatrix
  * @return the vector mean \f$\bar{a}_i\f$ or matrix mean \f$\bar{m}_{i,j}\f$
  */
-template <typename T, template <typename> class Container>
-typename linalg::int2float<T>::floatType mean(const Container<T>& a)
+template<typename T, template <typename> class Container>
+typename std::enable_if<std::is_same<T, complex128_t>::value, complex128_t>::type
+mean(const Container<T>& a)
 {
 	REQUIRE(a.size() > 0, "Vector/Matrix cannot be empty!\n");
 	return infer_backend(a)->mean(a);
 }
 
 /**
+ * Method that computes the mean of vectors or matrices composed of real numbers.
+ *
+ * @param a SGVector or SGMatrix
+ * @return the vector mean \f$\bar{a}_i\f$ or matrix mean \f$\bar{m}_{i,j}\f$
+ */
+template <typename T, template <typename> class Container>
+typename std::enable_if<!std::is_same<T, complex128_t>::value, float64_t>::type
+mean(const Container<T>& a)
+{
+	REQUIRE(a.size() > 0, "Vector/Matrix cannot be empty!\n");
+	return std::abs(infer_backend(a)->mean(a));
+}
+
```

Comment:
What I had here is calculating mean with only one method and cast the result to different type. But I realized that ViennaCL doesn't work that well with complex data and we'll have problem with colwise/rowwise mean operation.

Another way is to have independent `complex_mean` and `float_mean` in the backends implementation.